internal/tui/components/autocomplete/autocomplete.go

```diff
@@ -0,0 +1,270 @@
+	m.visible = false
+}
+
+func (m *Model) IsVisible() bool {
+	return m.visible
+}
+
+func (m *Model) SetPosition(x, y int) {
+	m.posX = x
+	m.posY = y
+}
+
+func (m *Model) SetWidth(width int) {
+	m.width = width
+}
+
+func (m *Model) View() string {
+	if !m.visible || len(m.filtered) == 0 {
+		return ""
+	}
+
+	numVisible := min(len(m.filtered), m.maxVisible)
+
+	var b strings.Builder
+
+	maxLabelWidth := m.width - 4
+
+	for i := 0; i < numVisible && i < len(m.filtered); i++ {
+		label := m.filtered[i]
+		if len(label) > maxLabelWidth {
+			label = label[:maxLabelWidth-3] + "..."
+		}
+
+		// Style based on selection
+		if i == m.selected {
+			// Selected row - use inverted colors
+			selectedStyle := lipgloss.NewStyle().
+				Background(m.ctx.Theme.SelectedBackground).
+				Foreground(m.ctx.Theme.InvertedText)
+			b.WriteString(selectedStyle.Render("❯ " + label))
+		} else {
+			// Non-selected row
+			b.WriteString("  " + label)
+		}
+
+		if i < numVisible-1 {
+			b.WriteString("\n")
+		}
+	}
+
+	popupStyle := lipgloss.NewStyle().
+		Border(lipgloss.RoundedBorder()).
+		BorderForeground(m.ctx.Theme.SecondaryBorder).
+		Foreground(m.ctx.Theme.PrimaryText).
+		Width(m.width)
+
+	return popupStyle.Render(b.String())
+}
+
+func extractCurrentLabel(input string) string {
+	lastComma := strings.LastIndex(input, ",")
+	if lastComma == -1 {
+		return input
+	}
+	return strings.TrimSpace(input[lastComma+1:])
+}
+
+type suggestionMatch struct {
+	label string
+	score float64
+}
+
+func filterAndRankSuggestions(suggestions []string, currentLabel string, excludeMap map[string]bool, maxResults int) []string {
+	currentLabelLower := strings.ToLower(currentLabel)
+
+	var matches []suggestionMatch
+
+	for _, suggestion := range suggestions {
+		if excludeMap[strings.ToLower(suggestion)] {
+			continue
+		}
+
+		suggestionLower := strings.ToLower(suggestion)
+
+		if strings.HasPrefix(suggestionLower, currentLabelLower) {
+			// Exact prefix match - highest score
+			matches = append(matches, suggestionMatch{
+				label: suggestion,
+				score: 1.0,
+			})
+		} else if strings.Contains(suggestionLower, currentLabelLower) {
+			// Contains match - lower score
+			matches = append(matches, suggestionMatch{
+				label: suggestion,
+				score: 0.5,
+			})
+		} else if currentLabelLower != "" && fuzzyMatch(suggestionLower, currentLabelLower) {
+			// Fuzzy match - lowest score
+			matches = append(matches, suggestionMatch{
+				label: suggestion,
+				score: 0.3,
+			})
+		}
+	}
+
+	// Sort by score descending, then alphabetically
+	sort.Slice(matches, func(i, j int) bool {
+		if matches[i].score != matches[j].score {
+			return matches[i].score > matches[j].score
+		}
+		return matches[i].label < matches[j].label
+	})
+
+	// Return top results
+	result := make([]string, 0, maxResults)
+	for i := 0; i < len(matches) && i < maxResults; i++ {
+		result = append(result, matches[i].label)
+	}
+
+	return result
+}
+
+func fuzzyMatch(text, pattern string) bool {
```

Comment:
I think it's more robust to go with a ready package for this.
Maybe we can copy what Charm did? Those guys are pretty good with go :D
https://github.com/charmbracelet/crush/blob/main/internal/tui/exp/list/filterable.go?plain=1#L251